src/components/deploymentConfig/DeploymentConfig.tsx

```diff
@@ -518,7 +518,7 @@ export default function DeploymentConfig({
             const deploymentTemplateResp = isProtected
                 ? await checkForProtectedLockedChanges()
                 : await getIfLockedConfigNonProtected(requestBody)
-            if (deploymentTemplateResp.result.isLockConfigError) {
+            if (deploymentTemplateResp.result?.isLockConfigError) {
                 setDisableSaveEligibleChanges(deploymentTemplateResp.result?.disableSaveEligibleChanges)
```

Comment:
@vivek-devtron Do I need to remove these optional chaining checks on line 552 and 553?